### DISCUSSION TITLE
"The requested resource wasn't found." when updating even after the row is created

### DISCUSSION THREAD

**Discussion by cs-clarence** [pocketbase/pocketbase#2657] [Q&A] [answered]
I have this function here that performs several creations:
```ts
async createuser(data: FormData) {
  let user: UsersResponse | null = null;
  if (!data.get("user")) {
// Success
    user = await userService.create<UsersResponse>({
      username: data.get("username") as string,
      email: data.get("email") as string,
      password: data.get("password") as string,
      passwordConfirm: data.get("confirmPassword") as string,
    });
  }

// Success
  const ep = await eprs.create<EmployeeProfilesResponse>(data);

  if (!user) {
    user = await usrs.getOne<UsersResponse>(data.get("user") as string);
  }

  if (!user.basicProfile) {
// Success
    const bp = await bprs.create<BasicProfilesResponse>(data);

// Fails here, as seen on the latest log below
    return await usrs.update(user.id, {
      employeeProfile: ep,
      basicProfile: bp,
    });
  }

  return await usrs.update(user.id, {
    employeeProfile: ep,
  });
},
```
However, the function fails at the last part where I update the user to attach some relations to it.

Here are the example logs during this transaction:
![image](https://github.com/pocketbase/pocketbase/assets/71768661/a86a62e3-0fec-4038-98d5-e7c158fa5849)

I don't know why it's failing on the patch request considering everything else succeeds.

**Comment by ganigeorgiev** [comment, marked as answer]
The API may return 404 error in the case the request client doesn't satisfy the Update API rule of the collection (eg. you may need to authenticate after creating the user if your collections have `@request.auth.* ...` rule).

**Reply by ashok-cse** [reply]
this is the response, i am getting and there is no api rule for CRUD ops

{
  "errorDetails": "sql: no rows in result set",
  "errorMessage": "The requested resource wasn't found."
}

**Reply by ganigeorgiev** [reply]
@ashok-cse Are you sure that you are passing the correct collection and record id when performing the update? Could you provide an example how you are sending the request?

**Reply by ashok-cse** [reply]
does it not work with collectionName and record id?

like this
/api/collections/designs/records/8js1ogfd59wnu1w

**Reply by ganigeorgiev** [reply]
@ashok-cse I'm not sure I follow. For the record update API you could check https://pocketbase.io/docs/api-records/#update-record.
If that doesn't help, please elaborate how are you sending the request and double check that the record with the provided id actually exist.

**Reply by ashok-cse** [reply]
I am using webapi/rest-api for crud ops

https://pb.mydomain.com/api/collections/designs/records/8js1ogfd59wnu1w

Using Patch method with formdata to perform update records

And I am following standard documentation

**Reply by ganigeorgiev** [reply]
@ashok-cse Again, double check that:
- the correct collection name is used
- the correct "designs" record id is used
- the Update API rule of the "designs" collection is empty OR the client request must satisfy it

Other than that I'm not really sure how to help you as I don't know how to reproduce it. If you think that it is a PocketBase issue, please create a **minimal** reproducible repo with sample data and I'll test it.

**Reply by ashok-cse** [reply]
Hey! Thanks & sorry 

I figured it out and resolve it

is there any docs to implement api rule using webapi/restapi and formdata?

**Reply by ganigeorgiev** [reply]
I don't understand what do you mean.
If you are asking for updating a collection configuration via web api you can refer to https://pocketbase.io/docs/api-collections/.

But in general if you want to create/update a collection programmatically I'd recommend using migrations:
- Go - https://pocketbase.io/docs/go-migrations/
- JS - https://pocketbase.io/docs/js-migrations/

If you have further questions, please consider filing a new discussion as it unnecessary spam the original author.
